akka-stream/src/main/protobuf/StreamRefContainers.proto

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+option java_package = "akka.stream.remote";
+option optimize_for = SPEED;
+
+/*************************************************
+  StreamRefs (SourceRef / SinkRef) related formats
+**************************************************/
+
+// TODO make explicit types for SinkRef/SourceRef as they can carry initial settings (initial demand etc)
+
+message SinkRef {
+    required ActorRef targetRef = 1;
+    optional int64 initialDemand = 2;
+}
+
+message ActorRef {
+    required string path = 1;
+}
+
+message Option {
+    optional Payload value = 1;
+}
+
+message Payload {
+    required bytes enclosedMessage = 1;
+    required int32 serializerId = 2;
+    optional bytes messageManifest = 4;
+}
+
+// stream refs protocol
+
+message CumulativeDemand {
+    required int64 seqNr = 1;
+}
+
+message SequencedOnNext {
+    required int64 seqNr = 1;
+    required Payload payload = 2;
+}
+
+message RemoteSinkFailure {
+    optional bytes cause = 1;
```

Comment:
is this intended for the full exception? that is difficult.
see https://github.com/akka/akka/blob/master/akka-remote/src/main/resources/reference.conf#L91-L96
and https://github.com/akka/akka/blob/master/akka-remote/src/main/scala/akka/remote/serialization/ThrowableSupport.scala

Perhaps it's enough to only serialize the exception message and use a general exception class for these failures?

Reply:
Yes, it's only the cause message indeed:

`  final case class RemoteSinkFailure(msg: String) extends StreamRefs.Protocol`

I don't think I'd want to transfer the entire stacktrace since the otherside can't do anything about it anyway, so just the message for informational purposes